pkg/reconciler/revision/revision.go

```diff
@@ -106,7 +106,9 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 	}
 
 	digests := make(chan digestData, len(rev.Spec.Containers))
-	for _, container := range rev.Spec.Containers {
+	containerIndexMap := map[string]int{}
+	for i, container := range rev.Spec.Containers {
+		containerIndexMap[container.Name] = i
```

Comment:
one possible approach:  since Go allows concurrently writing to an array from multiple goroutines as long as each individual element is only accessed from one, could we directly store to `rev.Status.ContainerStatus[i] = ..` in this loop (obviously making sure to do `i := i` and pre-create the array with the right size)? 

That way we can lose this map, the channel stuff and the second loop - and any errors could be returned normally via the `digestGrp.Wait()`.

Reply:
yep that as well :)